aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/customize/s3/S3Decorator.kt

```diff
@@ -64,6 +65,7 @@ class S3Decorator : ClientCodegenDecorator {
         setOf(
             ShapeId.from("com.amazonaws.s3#CopyObject"),
         )
+    private val operationsWithIsTruncatedPaginator = setOf(ShapeId.from("com.amazonaws.s3#ListPartsOutput"))
```

Comment:
**suggestion**: Processing the model and adding the trait could be it's own codegen decorator/integration. It isn't specific to S3 other than S3 has the only operation we know of right now with this, it could possibly be that other operations in the service fabric use something similar.

Reply:
Refactored to its own decorator in https://github.com/smithy-lang/smithy-rs/pull/3679/commits/a14a2eacf112eae221c3ca3abc47b34af6f360bb